### ISSUE TITLE
💅 `useAwait` warns incorrectly when an async function can return a Promise-like object

### ISSUE THREAD

**Issue by jpike88** [biomejs/biome#9291]
### Environment information

See below function;

```bash
class myClass extends someOtherAbstractClass {
    public async something(): Promise<boolean> {
    	if (someConditionMet) {
    		return anotherAsyncFunction();
    	}
    	return true;
    }
}    
```

`useAwait` forces me to make that function not async. But that would make the signature invalid, as well as deviates from the superclass definition.

I have many class instances that use that function, and some implementations of it use `await`, some may return another async function instead. There should be the ability to relax `useAwait` in these scenarios.

I would class this as a bug, as it's detrimental to any codebase that would use it, as it's too broad and destroys readability, consistency and is actually a serious risk of errors as certain async functions may be assumed incorrectly to behave like normal functions.

### Rule name

`useAwait`

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAcwBvAG0AZQBDAG8AbgBkAGkAdABpAG8AbgBNAGUAdAAgAD0AIAB0AHIAdQBlADsACgAKAGEAcwB5AG4AYwAgAGYAdQBuAGMAdABpAG8AbgAgAGEAbgBvAHQAaABlAHIAQQBzAHkAbgBjAEYAdQBuAGMAdABpAG8AbgAoACkAOgAgAFAAcgBvAG0AaQBzAGUAPABiAG8AbwBsAGUAYQBuAD4AIAB7AAoAIAAgAGMAbwBuAHMAdAAgAG0AeQBQAHIAbwBtAGkAcwBlACAAPQAgAG4AZQB3ACAAUAByAG8AbQBpAHMAZQA8AGIAbwBvAGwAZQBhAG4APgAoACgAcgBlAHMAbwBsAHYAZQApACAAPQA%2BACAAewAgAHIAZQBzAG8AbAB2AGUAKAB0AHIAdQBlACkAIAB9ACkAOwAKACAAIABjAG8AbgBzAHQAIAByAGUAcwB1AGwAdAAgAD0AIABhAHcAYQBpAHQAIABtAHkAUAByAG8AbQBpAHMAZQA7AAoAIAAgAHIAZQB0AHUAcgBuACAAcgBlAHMAdQBsAHQAOwAKAH0ACgAKAGMAbABhAHMAcwAgAE0AeQBDAGwAYQBzAHMAIAB7AAoACgAgACAAYQBzAHkAbgBjACAAdABoAGkAcwBTAGgAbwB1AGwAZABGAGEAaQBsACgAKQA6ACAAUAByAG8AbQBpAHMAZQA8AGIAbwBvAGwAZQBhAG4APgAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAdAByAHUAZQA7AAoAIAAgAH0ACgAKACAAIABhAHMAeQBuAGMAIABzAG8AbQBlAHQAaABpAG4AZwAoACkAOgAgAFAAcgBvAG0AaQBzAGUAPABiAG8AbwBsAGUAYQBuAD4AIAB7AAoAIAAgACAAIABpAGYAIAAoAHMAbwBtAGUAQwBvAG4AZABpAHQAaQBvAG4ATQBlAHQAKQAgAHsACgAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgAGEAbgBvAHQAaABlAHIAQQBzAHkAbgBjAEYAdQBuAGMAdABpAG8AbgAoACkAOwAKACAAIAAgACAAfQAKACAAIAAgACAAcgBlAHQAdQByAG4AIAB0AHIAdQBlADsACgAgACAAfQAKAH0ACgAKAGMAbwBuAHMAdAAgAHgAOgAgAE0AeQBDAGwAYQBzAHMAIAA9ACAAbgBlAHcAIABNAHkAQwBsAGEAcwBzACgAKQA7AAoAeAAuAHMAbwBtAGUAdABoAGkAbgBnACgAKQA7AA%3D%3D

### Expected result

The eslint version had the same problem, which got complaints:

https://github.com/eslint/eslint/pull/11167#issuecomment-446014215
https://github.com/eslint/eslint/discussions/18271#discussioncomment-14360805

This rule should be aligned with @typescript-eslint/require-await instead, which is a lot smarter.

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=MYewdgzgLgBBIFsCmBhcATAllT4CySsAvDFAE4CuSA3AFC0CGEAnmMDAGYVs7gwNgQUABZIyAQRZsAYt2C8wACgCUALhgAFMokwQkAHgBGIEABskAgHwwA3rRgxQkWAmZademCTBIA7pu0EXQNjMwswS0VFMiR4UwA3JGUvaxsYGLjExXIqZIBfZToHJ2h02IpTYn5fBmwYV3cgvSKyqAoyMDKICqg6PPpgUyYIGDxmFCGIEbt7GAAHCkNTTHYmVnYRXQBlYRAK9Gla0xV1RuCjE3MrW1mHGLaO0koaWf7ZhaWV-il2eGRNsAAcxOAQ8IUu4VStxgmA4MEUf1QGGwuDABCgyTsDmxrXanQEQlEEh%2Bsh4qJULQc-Rx9zxTyoLX6bxKYQAdJgwBwQIoxhNhoUgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aRAR1ko9oEMA7v0r50URNGgB7aJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
> some implementations of it use await, some may return another async function instead.

And how do you know that? How do you when to `await` the function?

**Comment by jpike88** [comment]
you assume that all of those implementations must be awaited in general, as per their superclass definition. There's no harm in needlessly `await`-ing a function... but the type drama above is also part of the issue for me
